### ISSUE TITLE
Errors at compilation

### ISSUE THREAD

**Issue by Contextualist** [Chion82/kcptun-raw#1]
I got the following error message on `make`:
```
trans_packet.c: In function 'check_packet_recv':
trans_packet.c:91:19: error: 'struct tcphdr' has no member named 'dest'
     if (ntohs(tcph->dest) != packetinfo->source_port) {
                   ^~
trans_packet.c:102:22: error: 'struct tcphdr' has no member named 'check'; did you mean 'th_ack'?
     pseudo_tcp_header->check = 0;
                      ^~
trans_packet.c:106:33: error: 'struct tcphdr' has no member named 'doff'; did you mean 'th_off'?
     int payloadlen = size - tcph->doff*4 - iphdrlen;
                                 ^~
trans_packet.c:123:29: error: 'struct tcphdr' has no member named 'check'; did you mean 'th_ack'?
     if (tcp_checksum != tcph->check) {
                             ^~
trans_packet.c:131:69: error: 'struct tcphdr' has no member named 'source'
     (*(packetinfo->on_packet_recv))(inet_ntoa(from_addr), ntohs(tcph->source), buffer + iphdrlen + tcph->doff*4, payloadlen, tcph->seq);
                                                                     ^~
trans_packet.c:131:104: error: 'struct tcphdr' has no member named 'doff'; did you mean 'th_off'?
     (*(packetinfo->on_packet_recv))(inet_ntoa(from_addr), ntohs(tcph->source), buffer + iphdrlen + tcph->doff*4, payloadlen, tcph->seq);
                                                                                                        ^~
trans_packet.c:131:130: error: 'struct tcphdr' has no member named 'seq'; did you mean 'th_seq'?
     (*(packetinfo->on_packet_recv))(inet_ntoa(from_addr), ntohs(tcph->source), buffer + iphdrlen + tcph->doff*4, payloadlen, tcph->seq);
                                                                                                                                  ^~
trans_packet.c: In function 'send_packet':
trans_packet.c:177:9: error: 'struct tcphdr' has no member named 'source'
     tcph->source = htons(packetinfo->source_port);
         ^~
trans_packet.c:178:9: error: 'struct tcphdr' has no member named 'dest'
     tcph->dest = sin.sin_port;
         ^~
trans_packet.c:179:9: error: 'struct tcphdr' has no member named 'seq'; did you mean 'th_seq'?
     tcph->seq = seq;
         ^~
trans_packet.c:180:9: error: 'struct tcphdr' has no member named 'ack_seq'; did you mean 'th_seq'?
     tcph->ack_seq = 0;
         ^~
trans_packet.c:181:9: error: 'struct tcphdr' has no member named 'doff'; did you mean 'th_off'?
     tcph->doff = 5;  //tcp header size
         ^~
trans_packet.c:182:9: error: 'struct tcphdr' has no member named 'fin'
     tcph->fin=0;
         ^~
trans_packet.c:183:9: error: 'struct tcphdr' has no member named 'syn'
     tcph->syn=1;
         ^~
trans_packet.c:184:9: error: 'struct tcphdr' has no member named 'rst'
     tcph->rst=0;
         ^~
trans_packet.c:185:9: error: 'struct tcphdr' has no member named 'psh'
     tcph->psh=0;
         ^~
trans_packet.c:186:9: error: 'struct tcphdr' has no member named 'ack'; did you mean 'th_ack'?
     tcph->ack=0;
         ^~
trans_packet.c:187:9: error: 'struct tcphdr' has no member named 'urg'
     tcph->urg=0;
         ^~
trans_packet.c:188:9: error: 'struct tcphdr' has no member named 'window'
     tcph->window = htons (5840); /* maximum allowed window size */
         ^~
trans_packet.c:189:9: error: 'struct tcphdr' has no member named 'check'; did you mean 'th_ack'?
     tcph->check = 0; //leave checksum 0 now, filled later by pseudo header
         ^~
trans_packet.c:190:9: error: 'struct tcphdr' has no member named 'urg_ptr'
     tcph->urg_ptr = 0;
         ^~
trans_packet.c:193:13: error: 'struct tcphdr' has no member named 'ack'; did you mean 'th_ack'?
         tcph->ack=1;
             ^~
trans_packet.c:209:9: error: 'struct tcphdr' has no member named 'check'; did you mean 'th_ack'?
     tcph->check = csum( (unsigned short*) pseudogram , psize);
         ^~
make[1]: *** [Makefile:7: build] Error 1
make: *** [Makefile:4: default] Error 2
```

**Comment by Chion82** [comment]
Hi. Please compile it on X86(_64) linux. I'm still working on this project and will provide pre-built releases later.

**Comment by Contextualist** [comment]
Thanks for reply. I did try that on x86_64 Alpine Linux. However, when I tried on a Ubuntu, it compiled successfully.

**Comment by Contextualist** [comment]
After googling I found way to get around this. The problem seems to be different standards of the platforms. Add `-D_GNU_SOURCE` macro flag for gcc, and it works well in Alpine Linux. I haven't test on other platforms.

**Comment by Chion82** [comment]
It may occur on some OpenWRT versions too. Thanks for your solution :)

**Comment by qq57240** [comment]
I had compiled it on arm and mipsel of openwrt successfully, but can't add libev.a into the binary by gcc.

**Comment by Chion82** [comment]
Fixed at #11
